src/ophys_etl/transforms/sine_dewarp.py

```diff
@@ -0,0 +1,606 @@
+import numpy as np
+import os
+import h5py
+import math
+import time
+import json
+import argparse
+import logging
+import multiprocessing
+import functools
+
+
+# Default parameters
+minvalue = 0.0
+maxlevel = 65535  # 16bit
+
+valid_nr_methods = [0, 1, 2, 3]
+
+
+def mode(data: np.ndarray):
+    """
+    Compute the mode of a numpy array.
+
+    Parameters
+    ----------
+    data: np.ndarray
+        Numpy array containing data of which we will calculate the mode.
+    Returns
+    -------
+    list
+        All values which appear the most number of times
+    int
+        The number of times the mode values appear
+    """
+    counts = {}
+
+    for x in data.flatten():
+        counts[x] = counts.get(x, 0) + 1
+
+    maxcount = max(counts.values())
+    modelist = []
+
+    for x in counts:
+        if counts[x] == maxcount:
+            modelist.append(x)
+
+    return modelist, maxcount
+
+
+def noise_reduce(data: np.array,
+                 xtable: dict,
```

Comment:
`xtable` seems like a great candidate for a `TypedDict` for a clear definition of what it is.